### PR TITLE
377 duplicate beacons

### PR DESCRIPTION
**Pull request by Evie-Skinner18**
## Context

- The Beacon Registry Team want to be able to identify duplicate beacon records and delete them

## Changes in this pull request

- New view to display hex IDs that match more than one beacon
- New view to display all the beacons for a given hex ID
- New DuplicatesController and Service
- New DuplicatesServiceUnitTests
- Additions to BeaconService, LegacyBeaconService and their repositories

## Link to Trello card
(https://trello.com/c/bLCzmZh2/377-identify-duplications-as-a-beacon-registry-team-member-i-can-identify-migrated-beacons-and-new-service-records-that-are-duplicat)

## Things to check
- Performance. Getting all the hex IDs that have more than one matching beacon in Java might be too expensive an operation
- We might need to do this as a SQL view

### CODE REVIEW

File: backoffice/src/components/duplicates/DuplicatesTables.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+      <MaterialTable
+        icons={tableIcons}
+        columns={columns}
+        data={(query) =>
+          new Promise(async (resolve, _reject) => {
+            const duplicateRows = await getPaginatedDuplicateSummaries(query);
+            resolve({
+              data: duplicateRows,
+              page: query.page,
+              totalCount: 23000,
```
Comment (slkendell-mt) [inline backoffice/src/components/duplicates/DuplicatesTables.tsx:115]: Why is this hard coded?

File: backoffice/src/components/layout/SettingsDrawer.tsx [hunk 2]
```diff
@@ -141,22 +142,6 @@ export const SettingsDrawer: React.FunctionComponent<ISettingsDrawerProps> = ({
               downloadStarted={handleDownloadStarted}
               downloadComplete={handleDownloadComplete}
             />
-            {/* TODO - Fix loading issue, remove button for now. */}
```
Comment (slkendell-mt) [inline backoffice/src/components/layout/SettingsDrawer.tsx:144]: I know this is currently commented out, but probably worth leaving in for now as they'll want it again soon

File: service/src/main/java/uk/gov/mca/beacons/api/beacon/application/BeaconService.java [hunk 3]
```diff
@@ -124,7 +119,33 @@ private ModelPatcher<Beacon> getPatcher() {
       .withMapping(Beacon::getModel, Beacon::setModel);
   }
 
-  public List<Beacon> findAll() {
-    return beaconRepository.findAll();
+  public Map<String, Integer> findHexIdsWithDuplicates(
+    int batchSize,
+    int numberAlreadyTaken
+  ) {
+    List<Beacon> batchOfBeacons = getBatch(batchSize, numberAlreadyTaken);
```
Comment (slkendell-mt) [inline service/src/main/java/uk/gov/mca/beacons/api/beacon/application/BeaconService.java:126]: Curious we're getting all the beacons here, do we not only care about what the duplicate factor is (hexId, and how many of them there are?)

File: service/src/main/java/uk/gov/mca/beacons/api/duplicates/application/DuplicatesService.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+      .limit(numberPerPage)
+      .collect(Collectors.toList());
+  }
+
+  public List<DuplicateBeaconDTO> getDuplicatesForHexId(String hexId) {
+    List<DuplicateBeacon> duplicateBeaconsForHexId = new ArrayList<>();
+    List<DuplicateBeaconDTO> duplicateBeaconDTOs = new ArrayList<>();
+
+    try {
+      List<Beacon> duplicateModernBeacons = beaconService.findByHexId(hexId);
```
Comment (slkendell-mt) [inline service/src/main/java/uk/gov/mca/beacons/api/duplicates/application/DuplicatesService.java:67]: This implies that if a hexID exists in modern, it won't exist in legacy. 

Real life scenario is more likely:

They had a legacy registration, now they have a modern one.

File: service/src/test/java/uk/gov/mca/beacons/api/legacybeacon/application/LegacyBeaconServiceUnitTest.java [hunk 5]
```diff
@@ -198,4 +201,92 @@ void delete_shouldUpdateTheNestedBeaconJsonObjectsLastModifiedDateToTodaysDate()
...
+      0
+    );
+    int numberOfDuplicatesForFirstHexId = duplicateHexIds
+      .values()
+      .stream()
+      .collect(Collectors.toList())
+      .get(0);
+
+    // assert
+    Assertions.assertEquals(2, duplicateHexIds.size());
```
Comment (Evie-Skinner18) [inline service/src/test/java/uk/gov/mca/beacons/api/legacybeacon/application/LegacyBeaconServiceUnitTest.java:257]: We left this with a need for pagination in the API endpoint that gets duplicate beacons. The story is currently in 'Potentials for Next SoW to order'

**Pull request closed without merging**